### PR TITLE
round end credits

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request
Round end credits. Port from Bay12

https://github.com/Bubberstation/Bubberstation/assets/77420409/5e1180b0-8c77-4cc7-8124-198e00d5e428

## Why It's Good For The Game

Content and it's cool.
## Changelog

:cl:
add: Round end credits
/:cl:

### CODE REVIEW

**Comment by StrangeWeirdKitten** [comment]
Actually based. I'll look at the code and we can TM it

File: modular_zubbers/code/modules/credits/credits.dm [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+GLOBAL_LIST(end_titles)
+#define JOINTEXT(X) jointext(X, null)
+/datum/controller/subsystem/ticker/declare_completion()
+	for(var/client/C)
+		if(!C.credits)
+			C.RollCredits()
+	for(var/thing in GLOB.clients)
+		var/client/C = thing
+		if (!C)
+			continue
```
Comment (Cyprex) [inline modular_zubbers/code/modules/credits/credits.dm:16]: Uh, this second for loop just does nothing as far as I can tell?

File: modular_zubbers/code/modules/credits/credits.dm [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+	. = ..()
+
+/datum/controller/subsystem/ticker/show_roundend_report(client/C, report_type = null)
+	. = ..()
+	for(var/iteration in GLOB.clients)
+		var/client/clients = iteration
+		if(!clients.credits)
+			clients.RollCredits()
```
Comment (Cyprex) [inline modular_zubbers/code/modules/credits/credits.dm:24]: Wait you already RollCredits() above in declare_completion() why here too?

File: modular_zubbers/code/modules/credits/credits.dm [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+/* /client
+	var/list/credits */
+
+
+/client/proc/RollCredits()
+	set waitfor = FALSE
+
+
+	if(!GLOB.end_titles)
+		GLOB.end_titles = generate_titles()
```
Comment (Cyprex) [inline modular_zubbers/code/modules/credits/credits.dm:34]: These should be generated before any calls to RollCredits() unless there's a good reason not to.

File: modular_zubbers/code/modules/credits/credits.dm [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+	sleep(CREDIT_ROLL_SPEED - CREDIT_SPAWN_SPEED)
+
+	ClearCredits()
+	verbs -= /client/proc/ClearCredits
+
+/client/proc/ClearCredits()
+	set name = "Stop End Titles"
+	set category = "OOC"
+	verbs -= /client/proc/ClearCredits
+	QDEL_NULL(credits)
```
Comment (Cyprex) [inline modular_zubbers/code/modules/credits/credits.dm:60]: This proc may be invoked twice- I don't know if that's an issue, I imagine not, but I'm noting it here anyways.

**Comment by Cyprex** [comment]
Reviewed the first half or so of this PR

**Comment by Majkl-J** [comment]
Stale for a month, reopen when work continues

**Pull request merged**